### DISCUSSION TITLE
REST API v3: API to add comment and label on to the pull request without having the file name

### DISCUSSION THREAD

**Discussion by RashmiRamanna** [community/community#24383] [Apps, API and Webhooks]
<p>I see APIs for adding comments and label for issues.</p>
<p>There is an API to post comment on pull request with file name as input.</p>
<p>I am looking for some API using which I can add label and comment on to the Pull Request just by having Pull Request number (just not specific to commit, file or line number)</p>
<p>I would appreciate if you point out to any such APIs.</p>

**Comment by lee-dohm** [comment]
<p>From the documentation:</p>
<blockquote>
<p>The Issue Comments API supports listing, viewing, editing, and creating comments on issues and pull requests.</p>
</blockquote>
<p> So use the standard API for <a href="https://developer.github.com/v3/issues/comments/#create-a-comment" rel="nofollow noopener">creating a comment</a>.</p>
<p>I hope that helps!</p>

**Comment by c-bata** [comment]
<p>Hi <a class="mention" href="/lee-dohm">@lee-dohm</a> </p>
<p>I bumped into the same problem. As you said, the document you linked says:</p>
<p>&gt; “The Issue Comments API supports listing, viewing, editing, and creating comments on issues and pull requests.”</p>
<p>&gt; <a href="https://developer.github.com/v3/issues/comments/#create-a-comment" rel="nofollow noopener">https://developer.github.com/v3/issues/comments/#create-a-comment</a></p>
<p>But there is no description about “how to comment on pull requests”, just includes API to post a comment on issues (POST /repos/:owner/:repo/issues/:issue_number/comments).</p>
<p>I tried following API.</p>
<p>* <code>POST /repos/:owner/:repo/issues/:pull_umber/comments</code></p>
<p>* <code>POST /repos/:owner/:repo/pulls/:pull_number/comments</code></p>

**Comment by c-bata** [comment]
<p>Sorry I resolved. I succeeded to a regular comment on PR by:</p>
<p>POST /repos/:owner/:repo/issues/:pull_number/comments</p>

**Comment by FirePing32** [comment]
<p>Hi <a class="mention" href="/c-bata">@c-bata</a> . How did you comment on the PR ? (I mean how did you authenticate yourself with the auth token). Can you paste the code block here for authentication of user with auth token…</p>

**Comment by c-bata** [comment]
<p>I just send a http request using curl with Authorization header.</p>

**Comment by FirePing32** [comment]
<p>Can you tell how can I do it with python ? I tried many ways and searched on web but could not find any helpful reference.</p>

**Comment by FirePing32** [comment]
<p><a class="mention" href="/c-bata">@c-bata</a>  Can you tell how can I do it with python ? I tried many ways and searched on web but could not find any helpful reference.</p>

**Comment by sayakpaul** [comment, marked as answer]
<p><a href="https://stackoverflow.com/questions/16744069/create-comment-on-pull-request" rel="nofollow noopener">This thread</a> might be useful.</p>
